api/src/test/java/org/pmiops/workbench/api/OfflineUserControllerTest.java

```diff
@@ -0,0 +1,103 @@
+package org.pmiops.workbench.api;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.db.dao.UserService;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@Import(LiquibaseAutoConfiguration.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class OfflineUserControllerTest {
+
+  @TestConfiguration
+  @Import({
+      OfflineUserController.class
+  })
+  static class Configuration {
+    @MockBean
```

Comment:
nit: This is interesting - I didn't even know you could do a mock this way. I think the pattern we have most of the other locations is to put this in an annotation on the `Configuration` itself, e.g.

https://github.com/all-of-us/workbench/blob/f3341eaf4b0988806120310bdac26e5d6b1b54b7/api/src/test/java/org/pmiops/workbench/api/AuditControllerTest.java#L56

Reply:
Gotcha, thanks for the pointer. Done.